pkg/orchestrator/assetscanestimationwatcher/watcher.go

```diff
@@ -160,13 +160,60 @@ func (w *Watcher) Reconcile(ctx context.Context, event AssetScanEstimationReconc
 			return err
 		}
 	case models.AssetScanEstimationStateStateFailed, models.AssetScanEstimationStateStateDone:
-		fallthrough
+		if err = w.reconcileDone(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
 	default:
 	}
 
 	return nil
 }
 
+func (w *Watcher) reconcileDone(ctx context.Context, assetScanEstimation *models.AssetScanEstimation) error {
+	if assetScanEstimation.EndTime == nil {
+		// TODO error?
+		return nil
+	}
+	if assetScanEstimation.TtlSecondsAfterFinished == nil {
+		// TODO error?
```

Comment:
why not just use the default?

Reply:
I can, but this might hide something not good that happened, since it should have been set before

Reply:
not sure what is better, if you just return a `nil` the object can stay in reconciling loop forever.
You can print an error and set the defaults, this way you will not hide anything

Reply:
Ideally the reconcile loop should fix things like this if it notices it, probably outside of any specific state processing. I.e. 
```
reconcile() {

if assetScanEstimation.TtlSecondsAfterFinished == nil {
   set default
}

set any other missing defaults

patch assetScanEstimation

switch assetScanEstimation.State {
 case pending:
    reconcilePending(...)
 case in progress:
    ...
 case done:
   ...
}

}
```